lib/src/main/java/com/cloudinary/android/preprocess/VideoDecoder.java

```diff
@@ -0,0 +1,71 @@
+package com.cloudinary.android.preprocess;
+
+import android.content.Context;
+import android.net.Uri;
+
+import com.cloudinary.android.payload.FilePayload;
+import com.cloudinary.android.payload.LocalUriPayload;
+import com.cloudinary.android.payload.Payload;
+import com.cloudinary.android.payload.PayloadNotFoundException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Returns the decoded video uri from a given payload. Payloads must be either {@link LocalUriPayload} or {@link FilePayload}.
+ * Note: It doesn't do the actual decoding process.
+ */
+public class VideoDecoder implements ResourceDecoder<Uri> {
+
+    /**
+     * Returns the video uri.
+     *
+     * @param context Android context.
+     * @param payload Payload to extract the resource from
+     * @throws PayloadDecodeException if the payload is neither a {@link LocalUriPayload} nor {@link FilePayload}
+     * @throws PayloadNotFoundException if the payload's resource cannot be found.
+     */
+    @Override
+    public Uri decode(Context context, Payload payload) throws PayloadDecodeException, PayloadNotFoundException {
+        File file;
+        if (payload instanceof LocalUriPayload) {
+            file = new File(context.getFilesDir() + "/tempFile_" + System.currentTimeMillis());
```

Comment:
I see why this was needed with the current API, but let's just change the API and reduce the complexity - add a  `T getData()` method on the `Payload` class, and then instead of writing to a new file just get the Uri directly, e.g.
`if (payload instanceof LocalUriPayload) {
    uri = ((LocalUriPayload)payload).getData();
}...`